pygmt/src/binstats.py

```diff
@@ -0,0 +1,122 @@
+"""
+binstats - Bin spatial data and determine statistics per bin
+"""
+from pygmt.clib import Session
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+from pygmt.io import load_dataarray
+
+
+@fmt_docstring
+@use_alias(
+    C="statistic",
+    E="empty",
+    G="outgrid",
+    I="spacing",
+    N="normalize",
+    R="region",
+    S="search_radius",
+    T="tiling",
+    V="verbose",
+    W="weight",
+    a="colname",
+    b="binary",
+    h="header",
+    i="incols",
+    r="registration",
+)
+@kwargs_to_strings(I="sequence", R="sequence", i="sequence_comma")
+def binstats(data, **kwargs):
+    r"""
+    Bin spatial data and determine statistics per bin.
+
+    Reads arbitrarily located (x,y[,z][,w]) points
+    (2-4 columns) from ``data``] and for each
+    node in the specified grid layout determines which points are
+    within the given radius.  These point are then used in the
+    calculation of the specified statistic. The results may be
+    presented as is or may be normalized by the circle area to
+    perhaps give density estimates.  Alternatively, select
+    hexagonal tiling instead or a rectangular grid layout.
```

Comment:
I think this is a typo in the GMT documentation https://docs.generic-mapping-tools.org/dev/gmtbinstats.html:
```suggestion
    hexagonal tiling instead of a rectangular grid layout.
```